### PR TITLE
Improve BML cleaning

### PR DESCRIPTION
**Pull request by lentzi90**
- Power off BMHs before tearing down the source cluster
- Remove local container registry
- Prune containers, volumes and images

~~This depends on https://github.com/metal3-io/metal3-dev-env/pull/887~~

### CODE REVIEW

**Comment by lentzi90** [comment]
After some more thinking I have changed this to simply power off the BMHs instead of deprovisioning. My reasoning is that deprovisioning will anyway happen at the end of a *successful* pipeline. If the pipeline is *not* successful, it is hard to say anything about what state the cluster and the machines are in. This means that there is a large risk that deprovisioning will fail anyway and could then block the pipeline. So instead of waiting on deprovisioning, simply power off.

**Review by Rozzii — APPROVED** [review]

File: jenkins/scripts/bare_metal_lab/deploy-lab.yaml [hunk 1]
```diff
@@ -56,19 +57,43 @@
         version: master
       tags: git
 
+    - name: Power off BMHs
```
Comment (fmuyassarov) [inline jenkins/scripts/bare_metal_lab/deploy-lab.yaml:60]: nit: this was a bit confusing for me, because it is not yet a BMH (k8s CRD) but actual server.

**Review by fmuyassarov — COMMENTED** [review]
/approve
/hold 
@lentzi90 feel free to unhold, just didn't wanted the PR to get merged before you saw the comment.

File: jenkins/scripts/bare_metal_lab/deploy-lab.yaml (hunk 1 above)
Comment (lentzi90) [reply jenkins/scripts/bare_metal_lab/deploy-lab.yaml:60]: Right, I understand the confusion. I was thinking that the point is really to make sure that the BMHs from the previous pipeline are cleaned up. At this point the management cluster is still running with these BMHs registered, so in that sense they are BMHs.

**Pull request merged**